sdk/src/feathers/text/VectorTextEditor.ls

```diff
@@ -72,101 +89,98 @@ package feathers.text
+                        panStage();
+                        closed = true;
+                        _keyboardState = KEYBOARD_CLOSED;
+                    }
+                    
                     // if we're the current editor AND we have focus, clear it
-                    if (_currentEditor == this && _hasIMEFocus)
-                    {                        
+                    if (_currentEditor == this)
+                    {
+                        //trace("Native keyboard closed, clearing focus");
+                        panStage();
                         clearFocus();
-                        _currentEditor = null;
+                        _keyboardState = KEYBOARD_CLOSED;
+                    }
+
+                    if (closed && _pendingEditor) {
+                        var editor = _pendingEditor;
+                        _pendingEditor = null;
+                        editor.setFocus();
                     }
                 }
-                else if (_hasIMEFocus)
-                {                        
+                else if (_currentEditor == this)
+                {
+                    _keyboardState = KEYBOARD_OPENED;
 
                     // resize is in device points, so we need to scale by stageHeight
-                    var scale = stage.nativeStageHeight / stage.stageHeight;
+                    var scale = stage.stageHeight / stage.nativeStageHeight;
                     resize = (resize) * scale;
-
+                    
                     // find the bounds of the text edit field, used to test if 
                     // we're obscured or not (and thus need to scroll)
                     var bounds = getBounds(stage);
-
+                    
                     // detect whether we need to scroll
-                    if ((stage.height - bounds.bottom) < (resize + 16))
+                    if ((stage.stageHeight - bounds.bottom) < (resize + _keyboardSafeZone))
                     {
                         // we need to scroll!  We do this with some animation
                         // so the user's eye can follow what is happening
-                        stage.addEventListener(Event.ENTER_FRAME, enterFrameHandler);
-                        _stageTargetY = (-resize) + (stage.height - bounds.bottom);
-
-                        // 16 pixel "safe zone"
-                        _stageTargetY -= height + 16; 
+                        panStage(bounds, resize, scale);
+                    } else {
+                        panStage();
                     }
-
-                    // and mark us as the current editor
-                    _currentEditor = this;
-
                 }
                  
-            }            
-
+            }
         }
-
-        protected function enterFrameHandler(event:Event):void
+        
+        protected function panStage(bounds:Rectangle = null, keyboardSize:Number = 0, scale:Number = 1)
```

Comment:
It's somewhat based on the text display itself (i.e. where/how the stage has to be moved to fit the display to the keyboard). We could move it into a common class, but I don't see too much of a value. If we wanted to support several types of editors it might be better to make a class that handles the tough stuff of handling focus and related stuff and abstract the behavior away more.